### PR TITLE
feat: remove SPORK_3_INSTANTSEND_BLOCK_FILTERING and SPORK_9_SUPERBLOCKS_ENABLED

### PR DESCRIPTION
**Pull request by UdjinM6**
## Issue being fixed or feature implemented
Both sporks have been effectively permanent fixtures of the network and their gating logic is no longer needed. 

## What was done?
Drop the enum entries, spork definitions, the `AreSuperblocksEnabled()` helper, and all associated branches in governance, InstantSend, masternode payments and mining RPC. `getblocktemplate` now always reports `superblocks_enabled` as `true`.

Functional tests are updated to stop toggling the removed sporks; `feature_multikeysporks.py` now uses `SPORK_19_CHAINLOCKS_ENABLED` as its second spork under test.

## How Has This Been Tested?
Run tests

## Breaking Changes
n/a

## Checklist:
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [ ] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Comment by thepastaclaw** [comment]
✅ Review complete (commit 18b990ba)

File: src/rpc/mining.cpp [hunk 1]
```diff
@@ -731,12 +731,10 @@ static RPCHelpMan getblocktemplate()
...
-    // next bock is a superblock and we need governance info to correctly construct it
-    CHECK_NONFATAL(node.sporkman);
-    if (AreSuperblocksEnabled(*node.sporkman)
-        && !node.mn_sync->IsSynced()
-        && CSuperblock::IsValidBlockHeight(active_chain.Height() + 1))
-            throw JSONRPCError(RPC_CLIENT_IN_INITIAL_DOWNLOAD, PACKAGE_NAME " is syncing with network...");
+    if (!node.mn_sync->IsSynced() && CSuperblock::IsValidBlockHeight(active_chain.Height() + 1)) {
+        // Next block is a superblock but we need governance info to correctly construct it.
+        throw JSONRPCError(RPC_CLIENT_IN_INITIAL_DOWNLOAD, PACKAGE_NAME " is syncing with network...");
+    }
```
Comment (thepastaclaw) [inline src/rpc/mining.cpp]: **🟡 Suggestion: Unconditional superblock sync check changes regtest getblocktemplate behavior**

SPORK_9's default value was `4070908800` (OFF). On regtest, `AreSuperblocksEnabled()` previously returned false unless the spork was explicitly activated, so this sync check was skipped. Now it runs unconditionally. On regtest, superblocks start at height 1500 with a 20-block cycle, and `CMasternodeSync::IsSynced()` is false unless the full sync state machine completes. This means `getblocktemplate` will start returning `RPC_CLIENT_IN_INITIAL_DOWNLOAD` every 20 blocks for plain regtest nodes that don't run masternode sync — a silent behavioral regression. Note: `generate()` (used by `BitcoinTestFramework`) is unaffected since it calls `BlockAssembler` directly, so functional tests won't catch this.

<details>
<summary>💡 Suggested change</summary>

```suggestion
    if (!Params().MineBlocksOnDemand()
        && !node.mn_sync->IsSynced()
        && CSuperblock::IsValidBlockHeight(active_chain.Height() + 1)) {
        // Next block is a superblock but we need governance info to correctly construct it.
        throw JSONRPCError(RPC_CLIENT_IN_INITIAL_DOWNLOAD, PACKAGE_NAME " is syncing with network...");
    }
```

</details>

<sub>source: ['codex']</sub>

<details>
<summary>🤖 Fix this with AI agents</summary>

```
These findings are from an automated code review. Verify each finding against the current code and only fix it if needed.

In `src/rpc/mining.cpp`:
- [SUGGESTION] lines 734-737: Unconditional superblock sync check changes regtest getblocktemplate behavior
  SPORK_9's default value was `4070908800` (OFF). On regtest, `AreSuperblocksEnabled()` previously returned false unless the spork was explicitly activated, so this sync check was skipped. Now it runs unconditionally. On regtest, superblocks start at height 1500 with a 20-block cycle, and `CMasternodeSync::IsSynced()` is false unless the full sync state machine completes. This means `getblocktemplate` will start returning `RPC_CLIENT_IN_INITIAL_DOWNLOAD` every 20 blocks for plain regtest nodes that don't run masternode sync — a silent behavioral regression. Note: `generate()` (used by `BitcoinTestFramework`) is unaffected since it calls `BlockAssembler` directly, so functional tests won't catch this.
```
</details>

File: test/functional/test_framework/test_framework.py [hunk 2]
```diff
@@ -1846,7 +1846,6 @@ def setup_network(self):
 
         # Enable InstantSend (including block filtering) and ChainLocks by default
```
Comment (thepastaclaw) [inline test/functional/test_framework/test_framework.py]: **💬 Nitpick: Stale comment still references block filtering as a spork-gated feature**

The comment says `# Enable InstantSend (including block filtering) and ChainLocks by default` but SPORK_3_INSTANTSEND_BLOCK_FILTERING was removed — block filtering is now always active. The parenthetical is misleading.

<details>
<summary>💡 Suggested change</summary>

```suggestion
        # Enable InstantSend and ChainLocks by default
```

</details>

<sub>source: ['claude']</sub>

**Review by thepastaclaw — COMMENTED** [review]
## Code Review

Clean spork removal with correct logic simplification. The main substantive issue is a behavioral regression in `getblocktemplate` on regtest: SPORK_9's default was OFF, so removing the gate makes the superblock sync check unconditional, breaking regtest mining at superblock heights. Additionally, several dead references to `CSporkManager` and `spork.h` remain and should be cleaned up.

*Reviewed commit: ee908799*

🟡 2 suggestion(s) | 💬 5 nitpick(s)

### 4 additional findings

<details>
<summary>🟡 <b>suggestion:</b> Dead member: m_sporkman is no longer used in CMNPaymentsProcessor</summary>

`src/masternode/payments.h (lines 20-56)`

All `AreSuperblocksEnabled(m_sporkman)` calls were removed from `payments.cpp`, but four artifacts remain: the forward declaration `class CSporkManager;` (line 20), the member `const CSporkManager& m_sporkman` (line 41), the constructor parameter (line 54), and the caller in `chainhelper.cpp:27` that passes `sporkman`. Grep confirms zero references to `m_sporkman` or `sporkman` in `payments.cpp`. This dead code should be removed as part of this cleanup PR.

</details>
<details>
<summary>💬 <b>nitpick:</b> Unused #include <spork.h> after AreSuperblocksEnabled removal</summary>

`src/governance/governance.cpp (line 14)`

The `#include <spork.h>` was needed for `AreSuperblocksEnabled` which called `CSporkManager::IsSporkActive`. Grep confirms no remaining references to `IsSporkActive`, `SporkId`, or any `SPORK_` constant in `governance.cpp`. This include is dead.

</details>
<details>
<summary>💬 <b>nitpick:</b> Dead forward declaration: class CSporkManager</summary>

`src/governance/governance.h (line 43)`

The `class CSporkManager;` forward declaration was only needed for the removed `AreSuperblocksEnabled(const CSporkManager&)` function declaration. Grep confirms no other reference to `CSporkManager` in this header.

</details>
<details>
<summary>💬 <b>nitpick:</b> Unused #include <governance/governance.h></summary>

`src/rpc/mining.cpp (line 46)`

The only governance symbol used in this file after the removal is `CSuperblock::IsValidBlockHeight`, which is declared in `governance/classes.h` (already included at line 45). No `CGovernanceManager`, `CGovernanceObject`, or other `governance.h`-specific symbols remain. This include can likely be removed, though compile-time verification is recommended since transitive dependencies may exist.

</details>

<details>
<summary>🤖 Prompt for all review comments with AI agents</summary>

```
These findings are from an automated code review. Verify each finding against the current code and only fix it if needed.

In `src/rpc/mining.cpp`:
- [SUGGESTION] lines 734-737: Unconditional superblock sync check changes regtest getblocktemplate behavior
  SPORK_9's default value was `4070908800` (OFF). On regtest, `AreSuperblocksEnabled()` previously returned false unless the spork was explicitly activated, so this sync check was skipped. Now it runs unconditionally. On regtest, superblocks start at height 1500 with a 20-block cycle, and `CMasternodeSync::IsSynced()` is false unless the full sync state machine completes. This means `getblocktemplate` will start returning `RPC_CLIENT_IN_INITIAL_DOWNLOAD` every 20 blocks for plain regtest nodes that don't run masternode sync — a silent behavioral regression. Note: `generate()` (used by `BitcoinTestFramework`) is unaffected since it calls `BlockAssembler` directly, so functional tests won't catch this.

In `src/masternode/payments.h`:
- [SUGGESTION] lines 20-56: Dead member: m_sporkman is no longer used in CMNPaymentsProcessor
  All `AreSuperblocksEnabled(m_sporkman)` calls were removed from `payments.cpp`, but four artifacts remain: the forward declaration `class CSporkManager;` (line 20), the member `const CSporkManager& m_sporkman` (line 41), the constructor parameter (line 54), and the caller in `chainhelper.cpp:27` that passes `sporkman`. Grep confirms zero references to `m_sporkman` or `sporkman` in `payments.cpp`. This dead code should be removed as part of this cleanup PR.
```
</details>

File: test/functional/feature_governance_cl.py [hunk 3]
```diff
@@ -55,7 +55,6 @@ def run_test(self):
         self.sync_blocks()
         self.wait_for_chainlocked_block_all_nodes(self.nodes[0].getbestblockhash())
 
-        self.nodes[0].sporkupdate("SPORK_9_SUPERBLOCKS_ENABLED", 0)
         self.wait_for_sporks_same()
```
Comment (thepastaclaw) [inline test/functional/feature_governance_cl.py]: **💬 Nitpick: Orphaned wait_for_sporks_same() with no preceding spork change**

This `wait_for_sporks_same()` previously followed `sporkupdate("SPORK_9_SUPERBLOCKS_ENABLED", 0)`. With that update removed, there's no spork change to propagate — the DKG spork at line 51 already has its own `wait_for_sporks_same()` at line 52. This call is now a harmless but confusing no-op.

<sub>source: ['claude']</sub>

File: doc/release-notes-7278.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+Removed Sporks
+--------------
+
+* `SPORK_3_INSTANTSEND_BLOCK_FILTERING` and `SPORK_9_SUPERBLOCKS_ENABLED` have
+  been removed. The behaviours they gated (InstantSend conflicting-block
+  rejection and superblock payments) are now always enabled. These sporks will
```
Comment (knst) [inline doc/release-notes-7278.md]: > I think it should be mentioned they already has no meaning on mainnet.

SPORK_3_INSTANTSEND_BLOCK_FILTERING and SPORK_9_SUPERBLOCKS_ENABLED have been removed. These sporks were already effectively always enabled on mainnet but continued to gate behavior on test networks (testnet, regtest, and devnet). The associated functionality (InstantSend conflicting-block rejection and superblock payments) is now permanently enabled across all networks, and the sporks will no longer appear in the spork RPC output.

**Review by knst — COMMENTED** [review]
concept ACK but PR requires more cleaning up unused things, see 2ed420d6c9885ba4cf00e2393c242e91fdfec9cc

**Review by thepastaclaw — COMMENTED** [review]
## Code Review

Clean removal of permanently-active sporks SPORK_3 and SPORK_9 with correct cleanup. The only substantive issue is that removing the `AreSuperblocksEnabled()` guard in `getblocktemplate` introduces a silent behavioral regression on test chains, where the superblock sync check now fires unconditionally. One stale comment references the removed block filtering spork.

*Reviewed commit: 3d9e80e5*

🟡 1 suggestion(s) | 💬 1 nitpick(s)

<details>
<summary>🤖 Prompt for all review comments with AI agents</summary>

```
These findings are from an automated code review. Verify each finding against the current code and only fix it if needed.

In `src/rpc/mining.cpp`:
- [SUGGESTION] lines 732-735: Unconditional superblock sync check changes `getblocktemplate` behavior on test chains
  Previously, this check was guarded by `AreSuperblocksEnabled(*node.sporkman)`, which checked SPORK_9. On mainnet, SPORK_9 was hardened to active (value `0`), so the check always ran — no behavioral change there. But on regtest/testnet/devnet, SPORK_9 defaulted to `4070908800` (OFF), so this sync check **never executed**. Now it runs unconditionally.

On regtest (`nSuperblockCycle = 20`), any node that hasn't completed masternode sync will get `RPC_CLIENT_IN_INITIAL_DOWNLOAD` from `getblocktemplate` every 20 blocks. The fix in `feature_dip3_deterministicmns.py` (adding `wait_until(... IsSynced)`) confirms the author is aware, but this patches one test caller rather than addressing the API-level change.

The `generate()` RPC bypasses `getblocktemplate` (calls `BlockAssembler` directly), so most functional tests won't surface this, but external tooling or custom test setups using `getblocktemplate` on regtest/devnet will silently break. Consider either:
- Guarding with `!Params().IsTestChain()` to restore previous test-chain behavior, or
- Documenting this as an intentional behavioral change in the release notes
```
</details>

File: src/rpc/mining.cpp [hunk 5]
```diff
@@ -731,12 +729,10 @@ static RPCHelpMan getblocktemplate()
...
```
Comment (thepastaclaw) [inline src/rpc/mining.cpp]: **🟡 Suggestion: Unconditional superblock sync check changes `getblocktemplate` behavior on test chains**

Previously, this check was guarded by `AreSuperblocksEnabled(*node.sporkman)`, which checked SPORK_9. On mainnet, SPORK_9 was hardened to active (value `0`), so the check always ran — no behavioral change there. But on regtest/testnet/devnet, SPORK_9 defaulted to `4070908800` (OFF), so this sync check **never executed**. Now it runs unconditionally.

On regtest (`nSuperblockCycle = 20`), any node that hasn't completed masternode sync will get `RPC_CLIENT_IN_INITIAL_DOWNLOAD` from `getblocktemplate` every 20 blocks. The fix in `feature_dip3_deterministicmns.py` (adding `wait_until(... IsSynced)`) confirms the author is aware, but this patches one test caller rather than addressing the API-level change.

The `generate()` RPC bypasses `getblocktemplate` (calls `BlockAssembler` directly), so most functional tests won't surface this, but external tooling or custom test setups using `getblocktemplate` on regtest/devnet will silently break. Consider either:
- Guarding with `!Params().IsTestChain()` to restore previous test-chain behavior, or
- Documenting this as an intentional behavioral change in the release notes

<sub>source: ['claude', 'codex']</sub>

<details>
<summary>🤖 Fix this with AI agents</summary>

```
These findings are from an automated code review. Verify each finding against the current code and only fix it if needed.

In `src/rpc/mining.cpp`:
- [SUGGESTION] lines 732-735: Unconditional superblock sync check changes `getblocktemplate` behavior on test chains
  Previously, this check was guarded by `AreSuperblocksEnabled(*node.sporkman)`, which checked SPORK_9. On mainnet, SPORK_9 was hardened to active (value `0`), so the check always ran — no behavioral change there. But on regtest/testnet/devnet, SPORK_9 defaulted to `4070908800` (OFF), so this sync check **never executed**. Now it runs unconditionally.

On regtest (`nSuperblockCycle = 20`), any node that hasn't completed masternode sync will get `RPC_CLIENT_IN_INITIAL_DOWNLOAD` from `getblocktemplate` every 20 blocks. The fix in `feature_dip3_deterministicmns.py` (adding `wait_until(... IsSynced)`) confirms the author is aware, but this patches one test caller rather than addressing the API-level change.

The `generate()` RPC bypasses `getblocktemplate` (calls `BlockAssembler` directly), so most functional tests won't surface this, but external tooling or custom test setups using `getblocktemplate` on regtest/devnet will silently break. Consider either:
- Guarding with `!Params().IsTestChain()` to restore previous test-chain behavior, or
- Documenting this as an intentional behavioral change in the release notes
```
</details>

File: test/functional/test_framework/test_framework.py (hunk 2 above)
Comment (thepastaclaw) [inline test/functional/test_framework/test_framework.py]: **💬 Nitpick: Stale comment still references block filtering as a toggleable feature**

SPORK_3_INSTANTSEND_BLOCK_FILTERING was removed in this PR and block filtering is now permanently active. The parenthetical `(including block filtering)` implies it's a separately controlled feature, which is no longer the case.

<details>
<summary>💡 Suggested change</summary>

```suggestion
        # Enable InstantSend and ChainLocks by default
```

</details>

<sub>source: ['claude']</sub>

File: src/rpc/mining.cpp (hunk 5 above)
Comment (thepastaclaw) [inline src/rpc/mining.cpp]: **🟡 Suggestion: Unconditional superblock sync check changes `getblocktemplate` behavior on test chains**

Before this PR, the sync check at superblock heights was gated by `AreSuperblocksEnabled()` (SPORK_9), which was OFF on test chains — so this check never executed on regtest/testnet/devnet. Now it runs unconditionally.

On regtest (`nSuperblockCycle = 20`), any node that hasn't completed masternode sync will get `RPC_CLIENT_IN_INITIAL_DOWNLOAD` every 20 blocks. The test fix in `feature_dip3_deterministicmns.py` (adding `wait_until(... IsSynced)`) confirms this is a real behavioral change.

Notably, the IBD check at lines 722–730 is already guarded by `!Params().IsTestChain()`, establishing a pattern of leniency on test chains that this new check doesn't follow. If this is fully intentional (as the release notes suggest), no code change is needed — but external test harnesses or mining tooling using `getblocktemplate` on regtest/devnet may silently break.

<sub>source: ['claude']</sub>

<details>
<summary>🤖 Fix this with AI agents</summary>

```
These findings are from an automated code review. Verify each finding against the current code and only fix it if needed.

In `src/rpc/mining.cpp`:
- [SUGGESTION] lines 732-735: Unconditional superblock sync check changes `getblocktemplate` behavior on test chains
  Before this PR, the sync check at superblock heights was gated by `AreSuperblocksEnabled()` (SPORK_9), which was OFF on test chains — so this check never executed on regtest/testnet/devnet. Now it runs unconditionally.

On regtest (`nSuperblockCycle = 20`), any node that hasn't completed masternode sync will get `RPC_CLIENT_IN_INITIAL_DOWNLOAD` every 20 blocks. The test fix in `feature_dip3_deterministicmns.py` (adding `wait_until(... IsSynced)`) confirms this is a real behavioral change.

Notably, the IBD check at lines 722–730 is already guarded by `!Params().IsTestChain()`, establishing a pattern of leniency on test chains that this new check doesn't follow. If this is fully intentional (as the release notes suggest), no code change is needed — but external test harnesses or mining tooling using `getblocktemplate` on regtest/devnet may silently break.
```
</details>

File: doc/release-notes-7278.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  but continued to gate behavior on test networks (testnet, regtest, and devnet).
+  The associated functionality (InstantSend conflicting-block rejection and
+  superblock payments) is now permanently enabled across all networks, and the
+  sporks will no longer appear in the `spork` RPC output.
+
+Updated RPCs
+------------
+
+* `getblocktemplate` now always reports `superblocks_enabled` as `true`. The
+  field is retained for backwards compatibility.
```
Comment (thepastaclaw) [inline doc/release-notes-7278.md:15]: **💬 Nitpick: Release notes should mention `getblocktemplate` sync requirement on test chains**

The release notes document that `getblocktemplate` now always reports `superblocks_enabled` as `true`, and that the removed functionality is "permanently enabled across all networks." However, the practical consequence — that `getblocktemplate` now throws `RPC_CLIENT_IN_INITIAL_DOWNLOAD` at superblock heights on regtest/testnet/devnet when the node hasn't completed masternode sync — is not called out. This is a subtle but potentially breaking change for test harnesses.

<details>
<summary>💡 Suggested change</summary>

```suggestion
* `getblocktemplate` now always reports `superblocks_enabled` as `true`. The
  field is retained for backwards compatibility. On test networks (regtest,
  testnet, devnet), `getblocktemplate` now requires masternode sync to be
  complete at superblock heights; previously this check was gated by the
  now-removed `SPORK_9`.
```

</details>

<sub>source: ['claude']</sub>

**Review by thepastaclaw — COMMENTED** [review]
## Code Review

Clean, well-executed removal of SPORK_3 and SPORK_9. All references are properly cleaned up. Both agents correctly identified that unconditional superblock enablement changes behavior on test chains — Claude identified the `getblocktemplate` behavioral shift and documentation gap, while Codex identified that `generate*` RPCs lack the equivalent sync guard. All three findings are valid but low-impact, affecting only test-chain edge cases.

*Reviewed commit: b2cde07e*

🟡 2 suggestion(s) | 💬 1 nitpick(s)

### 1 additional finding

<details>
<summary>🟡 <b>suggestion:</b> `generate*` RPCs lack the superblock sync guard added to `getblocktemplate`</summary>

`src/rpc/mining.cpp (lines 152-160)`

The PR adds a sync guard in `getblocktemplate()` (line 732) that refuses template creation at superblock heights until masternode sync completes. But `generateBlocks()` calls `BlockAssembler::CreateNewBlock()` directly with no equivalent check.

When an unsynced node mines at a superblock height via `generatetoaddress`/`generatetodescriptor`:
1. `FillBlockPayments` (payments.cpp:329) finds no governance trigger (no data) → builds block WITHOUT superblock payments
2. Local validation passes because `IsBlockValueValid` (payments.cpp:235) and `IsBlockPayeeValid` (payments.cpp:283) both early-return `true` when `!m_mn_sync.IsSynced()`
3. Synced peers with a valid governance trigger will reject this block

`generateblock` also lacks the guard — while it calls `TestBlockValidity`, that runs the same validation which is lenient for unsynced nodes.

Before this PR, SPORK_9 being OFF on test chains meant superblocks were disabled entirely, so this divergence was unreachable. Now it's possible.

Practical impact is limited: this requires test chain + unsynced node + superblock height + existing governance trigger + `generate*` usage. But if a guard is warranted for `getblocktemplate`, the same logic applies here. Note that a proper fix needs the check **inside** the `while` loop (since height changes each iteration), not before it.

</details>

<details>
<summary>🤖 Prompt for all review comments with AI agents</summary>

```
These findings are from an automated code review. Verify each finding against the current code and only fix it if needed.

In `src/rpc/mining.cpp`:
- [SUGGESTION] lines 732-735: Unconditional superblock sync check changes `getblocktemplate` behavior on test chains
  Before this PR, the sync check at superblock heights was gated by `AreSuperblocksEnabled()` (SPORK_9), which was OFF on test chains — so this check never executed on regtest/testnet/devnet. Now it runs unconditionally.

On regtest (`nSuperblockCycle = 20`), any node that hasn't completed masternode sync will get `RPC_CLIENT_IN_INITIAL_DOWNLOAD` every 20 blocks. The test fix in `feature_dip3_deterministicmns.py` (adding `wait_until(... IsSynced)`) confirms this is a real behavioral change.

Notably, the IBD check at lines 722–730 is already guarded by `!Params().IsTestChain()`, establishing a pattern of leniency on test chains that this new check doesn't follow. If this is fully intentional (as the release notes suggest), no code change is needed — but external test harnesses or mining tooling using `getblocktemplate` on regtest/devnet may silently break.
- [SUGGESTION] lines 152-160: `generate*` RPCs lack the superblock sync guard added to `getblocktemplate`
  The PR adds a sync guard in `getblocktemplate()` (line 732) that refuses template creation at superblock heights until masternode sync completes. But `generateBlocks()` calls `BlockAssembler::CreateNewBlock()` directly with no equivalent check.

When an unsynced node mines at a superblock height via `generatetoaddress`/`generatetodescriptor`:
1. `FillBlockPayments` (payments.cpp:329) finds no governance trigger (no data) → builds block WITHOUT superblock payments
2. Local validation passes because `IsBlockValueValid` (payments.cpp:235) and `IsBlockPayeeValid` (payments.cpp:283) both early-return `true` when `!m_mn_sync.IsSynced()`
3. Synced peers with a valid governance trigger will reject this block

`generateblock` also lacks the guard — while it calls `TestBlockValidity`, that runs the same validation which is lenient for unsynced nodes.

Before this PR, SPORK_9 being OFF on test chains meant superblocks were disabled entirely, so this divergence was unreachable. Now it's possible.

Practical impact is limited: this requires test chain + unsynced node + superblock height + existing governance trigger + `generate*` usage. But if a guard is warranted for `getblocktemplate`, the same logic applies here. Note that a proper fix needs the check **inside** the `while` loop (since height changes each iteration), not before it.
```
</details>

**Review by knst — DISMISSED** [review]
utACK b2cde07e62e55b9edee3d9d78ac0fcfce5bd0b66

File: src/rpc/mining.cpp (hunk 5 above)
Comment (UdjinM6) [reply src/rpc/mining.cpp]: Probably makes sense... added 18b990ba75

File: doc/release-notes-7278.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+* `getblocktemplate` now requires a fully synced node at superblock heights.
+  This check is skipped on test networks. Previously it was gated by
+  `SPORK_9_SUPERBLOCKS_ENABLED` which was already always active on mainnet.
```
Comment (thepastaclaw) [inline doc/release-notes-7278.md:19]: **🟡 Suggestion: Release notes still omit that `sporkupdate` now rejects the removed spork names**

This note covers the removed sporks disappearing from `spork` output, but it still misses the other user-visible RPC change from dropping these entries out of `sporkDefs`: `sporkupdate` now resolves `SPORK_3_INSTANTSEND_BLOCK_FILTERING` and `SPORK_9_SUPERBLOCKS_ENABLED` to `SPORK_INVALID` and returns `RPC_INVALID_PARAMETER` (`src/rpc/node.cpp:218-220`). That is a compatibility change for any automation that still tries to toggle those sporks, so it should be called out alongside the existing RPC notes.

<sub>source: ['codex-general']</sub>

<details>
<summary>🤖 Fix this with AI agents</summary>

```
These findings are from an automated code review. Verify each finding against the current code and only fix it if needed.

In `doc/release-notes-7278.md`:
- [SUGGESTION] lines 4-19: Release notes still omit that `sporkupdate` now rejects the removed spork names
  This note covers the removed sporks disappearing from `spork` output, but it still misses the other user-visible RPC change from dropping these entries out of `sporkDefs`: `sporkupdate` now resolves `SPORK_3_INSTANTSEND_BLOCK_FILTERING` and `SPORK_9_SUPERBLOCKS_ENABLED` to `SPORK_INVALID` and returns `RPC_INVALID_PARAMETER` (`src/rpc/node.cpp:218-220`). That is a compatibility change for any automation that still tries to toggle those sporks, so it should be called out alongside the existing RPC notes.
```
</details>

**Review by thepastaclaw — COMMENTED** [review]
## Code Review

One release-note omission remains in the spork-removal stack: it documents the `spork` and `getblocktemplate` RPC changes, but not the matching `sporkupdate` compatibility change for the removed spork names.

*Reviewed commit: 18b990ba*

🟡 1 suggestion(s)

<details>
<summary>🤖 Prompt for all review comments with AI agents</summary>

```
These findings are from an automated code review. Verify each finding against the current code and only fix it if needed.

In `doc/release-notes-7278.md`:
- [SUGGESTION] lines 4-19: Release notes still omit that `sporkupdate` now rejects the removed spork names
  This note covers the removed sporks disappearing from `spork` output, but it still misses the other user-visible RPC change from dropping these entries out of `sporkDefs`: `sporkupdate` now resolves `SPORK_3_INSTANTSEND_BLOCK_FILTERING` and `SPORK_9_SUPERBLOCKS_ENABLED` to `SPORK_INVALID` and returns `RPC_INVALID_PARAMETER` (`src/rpc/node.cpp:218-220`). That is a compatibility change for any automation that still tries to toggle those sporks, so it should be called out alongside the existing RPC notes.
```
</details>

**Review by PastaPastaPasta — APPROVED** [review]
utACK 18b990ba750c705404b964c4639669e2f39771e5
